File: Scripts/01_descriptive_and_correlations.py
"""Descriptive statistics and Pearson/Spearman correlations for the L2 longitudinal study."""
import pandas as pd

def run_descriptives(data_path="Longitudinal_Data_120.csv"):
    df = pd.read_csv(data_path)
    cols = [c for c in df.select_dtypes(include="number").columns if c not in ("ID", "Learner_ID")]
    desc = df[cols].describe().T
    desc["skewness"] = df[cols].skew()
    desc["kurtosis"] = df[cols].kurtosis()
    print(f"Participants: {len(df)}
", df.head())
    print("
DESCRIPTIVES
", desc.round(3))
    for method in ("pearson", "spearman"):
        corr = df[cols].corr(method=method)
        print(f"
{method.upper()} CORRELATIONS
", corr.round(3))
        corr.to_csv(f"{method}_correlation_matrix.csv")
    desc.to_csv("descriptive_statistics.csv")

if __name__ == "__main__":
    run_descriptives()

File: Scripts/01_descriptive_and_correlations.R
# Descriptive statistics and Pearson correlations
if (!requireNamespace("pacman", quietly=TRUE)) install.packages("pacman")
pacman::p_load(tidyverse, psych, Hmisc)
df <- read.csv("Longitudinal_Data_120.csv")
cat("=== SAMPLE ===\n"); print(glimpse(df))
num_df <- df %>% select(where(is.numeric), -any_of(c("ID","Learner_ID")))
cat("=== DESCRIPTIVES ===\n"); print(round(psych::describe(num_df)[,c("n","mean","sd","median","min","max","skew","kurtosis")],3))
r <- Hmisc::rcorr(as.matrix(num_df), type="pearson")
cat("=== PEARSON R ===\n"); print(round(r$r,3)); cat("=== P VALUES ===\n"); print(round(r$P,3))
